### DISCUSSION TITLE
Modularity of the library

### DISCUSSION THREAD

**Discussion by mgomez-tubs** [iavofficial/IAVFrontendFramework#23] [Ideas]
I think this framework could greatly benefit from the modularity of its core components. The abstractions that wrap these libraries (which are present in almost every React app, such as React Router and React i18n) are strongly use case oriented and this limits their features and from a general point of view, the scalabilty of this project, and the ones using it. On example is in the routing system. The root layout can't be changed for a single page, because its a top level layout. There are also issues when using subroutes, which this library doesn't seem to support directly, leading to warnings in the console. And I think that the routing and the sidebar being merged into the same logical construct really goes against the concept of separation of concerns. 

React Router has been the default routing library for years, it has a good documentation and lots of examples for basically any use case are available on the internet. And its safe to say that because of certain decisions made by the remix team in the past, some developers might prefer another library framework (although this is more of a selling point or developer expectation of a react framework). It would be way simpler, if this library provided wrapper routes that add additional functionalites and integration, instead of attempting to take care of the whole routing system. Refine, which imo has similar goals as this library, does [this](https://refine.dev/docs/guides-concepts/routing/) and even supports Next.js and Remix besides React Router.  Other Wrapper Routes could be provided to provide different features such as authentication or layouts in a more idiomatic way.

The same goes for react-i18n, although here the issues are less problematic (missing types, nested keys, some rarely used feats, incompatibilty with storybook). These could again all be solved easily if the library just provided the translations it offers, instead of wrapping the whole library.

Please take this critic as the feedback it is. I am aware that the goals of this library might be different from my expectations, but maybe the thoughts of an outsider can provide helpful insights.

**Comment by Jonas-Krahn** [comment]
Thank you for your thoughts.

Actually, we also thought about an architectural revision. We will work on this in the upcoming year. Our aim will be to create a modular system which allows you to completely replace for example the navigation logic.
